package/MDAnalysis/coordinates/FHIAIMS.py

```diff
@@ -0,0 +1,333 @@
+        # unitcell
+        'box_triclinic': "lattice_vector {box[0]:16.12f} {box[1]:16.12f} {box[2]:16.12f}\nlattice_vector {box[3]:16.12f} {box[4]:16.12f} {box[5]:16.12f}\nlattice_vector {box[6]:16.12f} {box[7]:16.12f} {box[8]:16.12f}\n"
+    }
+
+    def __init__(self, filename, convert_units=True, n_atoms=None, **kwargs):
+        """Set up the FHIAIMS Writer
+
+        Parameters
+        -----------
+        filename : str
+            output filename
+        n_atoms : int (optional)
+            number of atoms
+
+        """
+        self.filename = util.filename(filename, ext='.in', keep=True)
+        self.n_atoms = n_atoms
+
+    def write(self, obj):
+        """Write selection at current trajectory frame to file.
+
+        Parameters
+        -----------
+        obj : AtomGroup or Universe or :class:`Timestep`
+
+        """
+        # write() method that complies with the Trajectory API
+
+        try:
+
+            # make sure to use atoms (Issue 46)
+            ag_or_ts = obj.atoms
+            # can write from selection == Universe (Issue 49)
+
+        except AttributeError:
+            if isinstance(obj, base.Timestep):
+                ag_or_ts = obj.copy()
+            else:
+                raise_from(
+                    TypeError("No Timestep found in obj argument"), None)
+
+        # Check for topology information
+        missing_topology = []
+        try:
+            names = ag_or_ts.names
+        except (AttributeError, NoDataError):
+            names = itertools.cycle(('X',))
+            missing_topology.append('names')
+
+        try:
+            atom_indices = ag_or_ts.ids
+        except (AttributeError, NoDataError):
+            atom_indices = range(1, ag_or_ts.n_atoms+1)
+            missing_topology.append('ids')
+
+        if missing_topology:
+            warnings.warn(
+                "Supplied AtomGroup was missing the following attributes: "
+                "{miss}. These will be written with default values. "
+                "Alternatively these can be supplied as keyword arguments."
+                "".format(miss=', '.join(missing_topology)))
+
+        positions = ag_or_ts.positions
+        try:
+            velocities = ag_or_ts.velocities
+            has_velocities = True
+        except (AttributeError, NoDataError):
+            has_velocities = False
+
+        with util.openany(self.filename, 'wt') as output_fhiaims:
+            # Lattice
+            try:  # for AtomGroup/Universe
+                tri_dims = obj.universe.coord.triclinic_dimensions
+            except AttributeError:  # for Timestep
+                tri_dims = obj.triclinic_dimensions
+            # full output
+            try:
+                box = tri_dims.flatten()
+                output_fhiaims.write(self.fmt['box_triclinic'].format(box=box))
+            except (NameError, UnboundLocalError):
+                pass  # no box information
```

Comment:
I don't get the logic here. Can you explain? My thinking is:

Presumably, you don't want to write `lattice_vector` lines if there's no box. Under which circumstances do you get a `NameError` or a `UnboundLocalError` here ? You should always be able to get this information. Both are exceptions that I associate with unclear code paths. We should be more explicit here (even though in general I do like Python's "it's easier to ask for forgiveness than for permission" approach with exceptions.

There's also an additional consideration: We are not yet consistent in how we flag "no PBC" (see #2698 ): at the moment, `dimensions` can either be `None` (which we will adapt as the standard following #2698) or `[0,0,0,0,0,0]` — either way, `triclinic_vectors` will return a 3x3 array of zeros and that will write a non-sensical output line (I think). Luckily, using `triclinic_vectors` avoids the current issue with #2698 but one needs to test for zero-vectors.

 I suggest something like the following where we check that at least one element of the triclinic box vectors is non-zero. Perhaps a better test would check that at least one in each row is non-zero or that the volume is greater than zero (`ts.volume`).
```suggestion
           # full output
           if np.any(tri_dims != 0):      
                output_fhiaims.write(self.fmt['box_triclinic'].format(box=tri_dims.flatten()))
```

If you wanted to check the volume then I would change the preceding code to give you the `ts` object instead of `tri_dims` and then work with `ts.tri_dims` and `ts.volume`.

Reply:
Totally right.  The `np.any(tri_dims != 0)` is fine.